src/annotator/anchoring/xpath.js

```diff
@@ -1,52 +1,113 @@
-import { findChild } from './xpath-util';
-
 export { xpathFromNode } from './xpath-util';
 
 /**
- * Finds an element node using an XPath relative to the document root.
+ * Return the `index`'th immediate child of `element` whose tag name is
+ * `nodeName` (case insensitive).
+ *
+ * @param {Element} element
+ * @param {string} nodeName
+ * @param {number} index
  */
-function nodeFromXPathFallback(xpath, root) {
-  const steps = xpath.substring(1).split('/');
-  let node = root;
-  steps.forEach(step => {
-    let [name, idx] = step.split('[');
-    idx = idx ? parseInt(idx.split(']')[0]) : 1;
-    node = findChild(node, name.toLowerCase(), idx);
-  });
-  return node;
+function nthChildOfType(element, nodeName, index) {
+  nodeName = nodeName.toUpperCase();
+
+  let matchIndex = -1;
+  for (let i = 0; i < element.children.length; i++) {
```

Comment:
I never looked too closely had findChild, but I'm now wondering if we need to directly iterate over the list of child nodes? I'm not sure a query is any faster, but could we reduce some lines of code by doing something like this?

```
try {
    return element.querySelectorAll(`:scope > ${nodeName}`)[index];
  } catch(e) {
    return null;
  }
```

Reply:
I did not know about `:scope` in `querySelector`. Yes, I think this code would work. I expect it would be somewhat slower as it is doing a bunch more work (generating a selector string, parsing it, then executing selector matching). What is the `try...catch` for here?